src/loader.rs

```diff
@@ -17,47 +17,48 @@ impl AssetLoader for LibGdxAtlasAssetLoader {
         &["libgdx.atlas"]
     }
 
-    fn load<'a>(
+    async fn load<'a>(
         &'a self,
-        reader: &'a mut bevy::asset::io::Reader,
+        reader: &'a mut bevy::asset::io::Reader<'_>,
         _settings: &'a Self::Settings,
-        load_context: &'a mut bevy::asset::LoadContext,
-    ) -> bevy::utils::BoxedFuture<'a, Result<Self::Asset, Self::Error>> {
-        Box::pin(async move {
-            let mut file = String::new();
-            reader.read_to_string(&mut file).await?;
-
-            let asset = AssetFile::new(file)?;
-
-            let image: Image = load_context
-                .load_direct(
-                    load_context
-                        .asset_path()
-                        .path()
-                        .parent()
-                        .unwrap()
-                        .join(asset.file),
-                )
-                .await?
-                .take()
-                .expect("expected image asset");
-
-            let mut layout = TextureAtlasLayout::new_empty(asset.size.as_vec2());
-            let mut files = HashMap::new();
-
-            for frame in asset.files {
-                let id = layout.add_texture(frame.bounds);
-                files.insert(frame.filename, id);
-            }
-
-            let atlas = load_context.add_labeled_asset("atlas_layout".into(), layout);
-            let image = load_context.add_labeled_asset("atlas_texture".into(), image);
-
-            Ok(LibGdxAtlasAsset {
-                image,
-                atlas,
-                files,
-            })
+        load_context: &'a mut bevy::asset::LoadContext<'_>,
+    ) -> Result<Self::Asset, Self::Error> {
+        let mut file = String::new();
+        reader.read_to_string(&mut file).await?;
+
+        let asset = AssetFile::new(file)?;
+
+        let path = load_context
+            .asset_path()
+            .path()
+            .parent()
+            .unwrap()
+            .join(asset.file);
+
+        let image: Image = load_context
+            .loader()
+            .direct()
+            .untyped()
+            .load(path)
+            .await?
+            .take()
+            .expect("expected image asset");
```

Comment:
this will still panic in the end. is that common practice in an `AssetLoader`? needs investigation or change to proper handlable error.
(fine in followup as this is already in the old version)